modules/porous_flow/src/userobjects/PorousFlowCount.C

```diff
@@ -0,0 +1,108 @@
+#include "PorousFlowCount.h"
+#include "MooseMesh.h"
+
+template <>
+InputParameters
+validParams<PorousFlowCount>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+
+  MultiMooseEnum setup_options(SetupInterface::getExecuteOptions());
+  setup_options = "timestep_begin";
+  params.set<MultiMooseEnum>("execute_on") = setup_options;
+  return params;
+}
+
+PorousFlowCount::PorousFlowCount(const InputParameters & parameters)
+  : ElementUserObject(parameters), _num_upwinds(0), _num_downwinds(0), _proto_flux(0)
+{
+}
+
+PorousFlowCount::~PorousFlowCount()
+{
+  // need to free _num_upwinds, etc ?
+}
+
+void
+PorousFlowCount::initialize()
+{
+  // this gets called once at the start of each timestep,
+  // before execute()
+  _num_upwinds.resize(_mesh.nElem());
+  _num_downwinds.resize(_mesh.nElem());
+  _proto_flux.resize(_mesh.nElem());
+}
+
+void
+PorousFlowCount::execute()
+{
+  // gets executed at the start of each timestep,
+  // due to the execute_on = timestep_begin option,
+  // for each element.
+  // the following resize is in preparation for setProtoFlux
+  _num_upwinds[_current_elem->id()].resize(0);
```

Comment:
Keep in mind that execute on an elemental Aux variable is called per quadrature point so this will get called several times. This isn't a problem with this logic, but if you do something more expensive it might be a little repetitive.